tests/pytests/unit/state/test_reactor_compiler.py

```diff
@@ -0,0 +1,669 @@
+import logging
+
+import pytest
+
+import salt.minion
+import salt.state
+from salt.utils.odict import OrderedDict
+from tests.support.mock import MagicMock, patch
+
+log = logging.getLogger(__name__)
+
+
+pytestmark = [
+    pytest.mark.core_test,
+]
+
+
+def test_compiler_render_template(minion_opts, tmp_path):
+    """
+    Test Compiler.render_template
+    """
+    minion = "poc-minion"
+    kwargs = {
+        "tag": f"salt/minion/{minion}/start",
+        "data": {
+            "id": minion,
+            "cmd": "_minion_event",
+            "pretag": None,
+            "data": f"Minion {minion} started at Thu Sep 14 07:31:04 2023",
+            "tag": "salt/minion/{minion}/start",
```

Comment:
```suggestion
            "tag": f"salt/minion/{minion}/start",
```
?
And if this is how it should look like, why are the tests passing?

Reply:
The code does not use this part of the kwargs dictionary. I still want to keep this kwarg as it is, since this is what the kwarg data looks like when running the reactor.

Reply:
> since this is what the kwarg data looks like when running the reactor.

`{minion}` ? the format string? not the actual minion ID?

Reply:
The actual data includes the minion ID. I updated the test to reflect that. Was just trying to answer your question as to why the tests where passing when the minion id was not in `data['tag']`

Reply:
Ah, makes sense now.